packages/notifications-controller/src/MetamaskNotificationsController.ts

```diff
@@ -0,0 +1,1142 @@
+  },
+};
+export const defaultState: MetamaskNotificationsControllerState = {
+  subscriptionAccountsSeen: [],
+  isMetamaskNotificationsFeatureSeen: false,
+  isMetamaskNotificationsEnabled: false,
+  isFeatureAnnouncementsEnabled: false,
+  metamaskNotificationsList: [],
+  metamaskNotificationsReadList: [],
+  isUpdatingMetamaskNotifications: false,
+  isFetchingMetamaskNotifications: false,
+  isUpdatingMetamaskNotificationsAccount: [],
+  isCheckingAccountsPresence: false,
+};
+
+export declare type MetamaskNotificationsControllerUpdateMetamaskNotificationsList =
+  {
+    type: `${typeof controllerName}:updateMetamaskNotificationsList`;
+    handler: MetamaskNotificationsController['updateMetamaskNotificationsList'];
+  };
+
+export declare type MetamaskNotificationsControllerDisableMetamaskNotifications =
+  {
+    type: `${typeof controllerName}:disableMetamaskNotifications`;
+    handler: MetamaskNotificationsController['disableMetamaskNotifications'];
+  };
+
+export declare type MetamaskNotificationsControllerSelectIsMetamaskNotificationsEnabled =
+  {
+    type: `${typeof controllerName}:selectIsMetamaskNotificationsEnabled`;
+    handler: MetamaskNotificationsController['selectIsMetamaskNotificationsEnabled'];
+  };
+
+// Messenger Actions
+export type Actions =
+  | MetamaskNotificationsControllerUpdateMetamaskNotificationsList
+  | MetamaskNotificationsControllerDisableMetamaskNotifications
+  | MetamaskNotificationsControllerSelectIsMetamaskNotificationsEnabled
+  | ControllerGetStateAction<'state', MetamaskNotificationsControllerState>;
+
+// Allowed Actions
+export type AllowedActions =
+  // Keyring Controller Requests
+  | KeyringControllerGetAccountsAction
+  // Auth Controller Requests
+  | AuthenticationControllerGetBearerToken
+  | AuthenticationControllerIsSignedIn
+  // User Storage Controller Requests
+  | UserStorageControllerEnableProfileSyncing
+  | UserStorageControllerGetStorageKey
+  | UserStorageControllerPerformGetStorage
+  | UserStorageControllerPerformSetStorage
+  // Push Notifications Controller Requests
+  | PushPlatformNotificationsControllerEnablePushNotifications
+  | PushPlatformNotificationsControllerDisablePushNotifications
+  | PushPlatformNotificationsControllerUpdateTriggerPushNotifications;
+
+// Events
+export type MetamaskNotificationsControllerMessengerEvents =
+  ControllerStateChangeEvent<
+    typeof controllerName,
+    MetamaskNotificationsControllerState
+  >;
+
+// Allowed Events
+export type AllowedEvents =
+  | KeyringControllerStateChangeEvent
+  | PushPlatformNotificationsControllerOnNewNotificationEvent;
+
+// Type for the messenger of MetamaskNotificationsController
+export type MetamaskNotificationsControllerMessenger =
+  RestrictedControllerMessenger<
+    typeof controllerName,
+    Actions | AllowedActions,
+    AllowedEvents,
+    AllowedActions['type'],
+    AllowedEvents['type']
+  >;
+
+/**
+ * Controller that enables wallet notifications and feature announcements
```

Comment:
What is the difference between this controller, `NotificationController`, and `AnnouncementController` (the last two already exist in this repo).

Reply:
R.E. Notifications are not unified.

- Notifications Controller was initially created by the snaps team for snap notifications. I believe there is state for notifications and their read/dismissed state

- Announcement Controller is new to me. Unsure how announcements differ from notifications (Maybe this is the startup modals you see as a new user opening the wallet?).

- Our controller contains a lot of defined logic for interfacing with our APIs to deliver specific metamask notifications:
  - Wallet Notifications: Send/Received tokens & nfts, Swaps, Staking.
  - Feature Announcements: Mostly for marketing new features in the wallet
  - (Future) Security Notifications: TBD.
  
I think we should unify these in the future for clearer consistency.